### PR TITLE
Add links to DALI success stories in README

### PR DESCRIPTION
**Pull request by JanuszL**
- gathers example DALI success stories from customers
  and known use cases and add them to the README

## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- gathers example DALI success stories from customers
  and known use cases and add them to the README

### Affected modules and functionalities:
- readme

### Key points relevant for the review:
- wording
- if no known example is missing

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [x] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: DALI-3275

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -118,6 +118,31 @@ Highlights
...
 
+Who is using DALI:
+------------------
+
+- |kaggle_dali_link|_: |kaggle_great_dali_link|_
+- |neuroscience_DALI|_
+- |HPC_DALI|_
+- |MLPERF_DALI|_
+
+.. |kaggle_dali_link| replace:: It is used during Kaggle computer vision competitions
```
Comment (szkarpinski) [inline README.rst:129]:
```suggestion
.. |kaggle_dali_link| replace:: During Kaggle computer vision competitions
```
Nitpick: It'll look better with all bullets in the same form

**Review by szkarpinski — APPROVED** [review]

File: README.rst (hunk 1 above)
Comment (JanuszL) [reply README.rst]: done

**Comment by JanuszL** [comment]
!build

**Review by szkarpinski — APPROVED** [review]

File: README.rst [hunk 2]
```diff
@@ -118,6 +118,31 @@ Highlights
...
+.. _kaggle_great_dali_link: https://www.kaggle.com/competitions/rsna-breast-cancer-detection/discussion/391059
+
+.. |neuroscience_DALI| replace:: As a tool that helps moving the needle in Neuroscience experiments
+.. _neuroscience_DALI: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10168383/
+
+.. |HPC_DALI| replace:: To improve the resource utilization in Advanced Computing Infrastructure
+.. _HPC_DALI: https://arcwiki.rs.gsu.edu/en/dali/using_nvidia_dali_loader
+
+.. |MLPERF_DALI| replace:: In the cutting edge industry benchmarks
+.. _MLPERF_DALI: https://developer.nvidia.com/blog/mlperf-hpc-v1-0-deep-dive-into-optimizations-leading-to-record-setting-nvidia-performance/
```
Comment (klecki) [inline README.rst:142]: How about having a format like:
* <Name of the project> - <explanation>
so, for example:
* Lightning Pose - state of the art pose estimation research model
for the 3rd item

Comment (JanuszL) [reply README.rst]: How about now?

**Review by klecki — APPROVED** [review]

**Comment by JanuszL** [comment]
!build

**Review by szkarpinski — APPROVED** [review]

**Pull request merged**